test/core/storage/trie/trie_db/trie_batch_test.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @given an empty trie
+ * @when putting some data to it
+ * @then inserted data is accessible
+ */
+
+#include "storage/trie/polkadot_trie_db/polkadot_trie_batch.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <testutil/storage/map_storage.hpp>
+#include "crypto/hasher/hasher_impl.hpp"
+#include "testutil/literals.hpp"
+#include "testutil/outcome.hpp"
+#include "testutil/storage/base_leveldb_test.hpp"
+#include "testutil/storage/polkadot_trie_db_printer.hpp"
+
+using namespace kagome::storage::trie;
+using kagome::common::Buffer;
+using kagome::hash::HasherImpl;
+using testing::_;
+using testing::Invoke;
+using testing::Return;
+
+class TrieBatchTest : public test::BaseLevelDB_Test {
+ public:
+  TrieBatchTest() : BaseLevelDB_Test("/tmp/leveldbtest") {}
+
+  void SetUp() override {
+    open();
+    trie = std::make_unique<PolkadotTrieDb>(std::move(db_));
+  }
+
+  std::vector<std::pair<Buffer, Buffer>> data = {
+      {"123456"_hex2buf, "42"_hex2buf},
+      {"1234"_hex2buf, "1234"_hex2buf},
+      {"010203"_hex2buf, "0a0b"_hex2buf},
+      {"010a0b"_hex2buf, "1337"_hex2buf},
+      {"0a0b0c"_hex2buf, "deadbeef"_hex2buf}};
+
+  std::unique_ptr<PolkadotTrieDb> trie;
+};
+
+class MockPolkadotTrieDb : public PolkadotTrieDb {
+ public:
+  MockPolkadotTrieDb() : PolkadotTrieDb(nullptr) {}
+
+  MOCK_CONST_METHOD0(getRootHash, Buffer());
+};
+
+class MockDb : public test::MapDb {
+ public:
+  MOCK_METHOD2(put, outcome::result<void>(const Buffer &, const Buffer &));
+
+  // to retain the ability to call the actual implementation of put from the
+  // superclass
+  outcome::result<void> true_put(const Buffer &key, const Buffer &value) {
+    return MapDb::put(key, value);
+  }
+};
+
+/**
+ * @given an empty trie
+ * @when putting some entries into it using a batch
+ * @then all inserted entries are accessible from the trie
+ */
+TEST_F(TrieBatchTest, Put) {
+  auto batch = trie->batch();
+  for (auto &entry : data) {
+    EXPECT_OUTCOME_TRUE_void(_, batch->put(entry.first, entry.second));
+  }
+  for (auto &entry : data) {
+    EXPECT_OUTCOME_TRUE(res, trie->get(entry.first));
+    ASSERT_TRUE(res.empty());
+  }
+  EXPECT_OUTCOME_TRUE_void(r, batch->commit());
+  for (auto &entry : data) {
+    EXPECT_OUTCOME_TRUE(res, trie->get(entry.first));
+    ASSERT_EQ(res, entry.second);
+  }
+
+  EXPECT_OUTCOME_TRUE_void(__, trie->put("102030"_hex2buf, "010203"_hex2buf));
+  EXPECT_OUTCOME_TRUE_void(_, trie->put("104050"_hex2buf, "0a0b0c"_hex2buf));
```

Comment:
You can just `ASSERT_TRUE` to check the outcome result without retrieving the value

Reply:
EXPECT_OUTCOME_TRUE also outputs an error message from the result, if one occurs, so it's more convenient.

Reply:
UPD: replaced with EXPECT_OUTCOME_TRUE_1